pkg/cli/codearea.go

```diff
@@ -245,21 +316,25 @@ func (w *codeArea) handleKeyEvent(key ui.Key) bool {
 		w.State.Buffer.InsertAtDot(s)
 		w.inserts += s
 		w.lastCodeBuffer = w.State.Buffer
-		var abbr, full string
-		// Try to expand an abbreviation, preferring the longest one
-		w.Abbreviations(func(a, f string) {
-			if strings.HasSuffix(w.inserts, a) && len(a) > len(abbr) {
-				abbr, full = a, f
-			}
-		})
-		if len(abbr) > 0 {
-			c := &w.State.Buffer
-			*c = CodeBuffer{
-				Content: c.Content[:c.Dot-len(abbr)] + full + c.Content[c.Dot:],
-				Dot:     c.Dot - len(abbr) + len(full),
-			}
-			w.resetInserts()
-		}
+		w.expandSimpleAbbr()
+		w.expandSmallWordAbbr(key.Rune, len(s))
 		return true
 	}
 }
+
+// Determine if the rune is an alphanumeric character.
+func IsAlnum(r rune) bool {
+	return unicode.IsLetter(r) || unicode.IsNumber(r)
+}
+
+// Determine if the rune is whitespace, alphanum, or something else.
+func CategorizeSmallWord(r rune) int {
+	switch {
+	case unicode.IsSpace(r):
+		return 0
+	case IsAlnum(r):
+		return 1
+	default:
+		return 2
+	}
+}
```

Comment:
This is fine, just noting that I'd like to move all the word logic from `pkg/edit` to `pkg/cli` in future.